jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -2255,18 +2261,39 @@ public boolean hasFullPrivilegeAccess() {
 			STRONG
 		}
 
-		static class ClassDefiner {
-			Class<?> defineClass(boolean option) {
-				return null;
+		static final class ClassDefiner {
+			private final byte[] classBytes;
+			private final String className;
+			private final Lookup lookup;
+			ClassDefiner(String name, byte[] template, Lookup lookupObj) {
+				className = name;
+				classBytes = template;
+				lookup = lookupObj;
+			}
+			Class<?> defineClass(boolean initOption) {
+				JavaLangAccess jlAccess = SharedSecrets.getJavaLangAccess();
+				Class<?> lookupClass = lookup.lookupClass();
+				return jlAccess.defineClass(lookupClass.getClassLoader(), lookupClass, className, classBytes, jlAccess.protectionDomain(lookupClass), initOption, 0, null);
```

Comment:
@hangshao0 pls wrap this in an AccessController.doPrivileged() to fix some of the test failures.

Reply:
Fixed.